### PR TITLE
Handling user logout and registering gql tag

### PR DESCRIPTION
**Pull request by tomitrescak**
This pull request handles:
* the removal of user from the context on server upon logout 
* registration of gal tag on client

### CODE REVIEW

**Comment by lorensr** [comment]
Thanks Tomas! I didn't add `user` to `context` because when we add `userId` caching #6, we won't always have fetched the `user`, and ideally we won't be fetching the user every time, so as to improve latency. I think the best way to get the user is #4.

File: main-client.js [hunk 1]
```diff
@@ -4,6 +4,10 @@ import { createNetworkInterface } from 'apollo-client';
 import { addTypenameToSelectionSet } from 'apollo-client/queries/queryTransform';
 import { Accounts } from 'meteor/accounts-base';
 import { _ } from 'meteor/underscore';
+import { registerGqlTag } from 'apollo-client/gql';
+
+// register gql tag for queries
+registerGqlTag();
```
Comment (lorensr) [inline main-client.js]: Stylistically I prefer not having comments that mainly repeat the code. In this case, I'd either assume the reader knows what a gql tag is or change the comment to maybe `// setup query template literal tag`, probably former

**Comment by tomitrescak** [comment]
Can you elaborate on #4 for user caching? I actually created buy own caching mechanism, but it's very silly. would love to have something more robust.

**Comment by lorensr** [comment]
Sure, something like this:

```coffee
USER_CACHE_MAX = 1000
cache = lruCache(config.userCacheMax || USER_CACHE_MAX)

// inside apollo options function
cacheEntry = cache.get req.headers.authorization
if cacheEntry.user 
  isExpired = cacheEntry.expiresAt < new Date()
  if not isExpired
    options.context = {user}
else
  updateCache = (user) ->
    cache.set req.headers.authorization,
      user: user
      expiresAt: // todo

  cursor = Meteor.users.find({"services.resume.loginTokens.hashedToken": hashedToken})
  user = await cursor.findOne()
  if user
    updateCache(user)
    options.context = {user}

    handle = cursor.observe
      changed(newDoc, oldDoc): ->
        if token expired
          cache.del(req.headers.authorization)
          handle.stop()
        else
          if cache.has req.headers.authorization
            updateCache(newDoc)
          else
            handle.stop()
      removed(oldDoc): ->
        cache.del(req.headers.authorization)
        handle.stop()
```

**Comment by lorensr** [comment]
Using https://www.npmjs.com/package/lru-cache

**Comment by tomitrescak** [comment]
@lorensr you are a star! Thank you!

**Comment by stubailo** [comment]
`registerGqlTag` is now deprecated, so we shouldn't use it anymore.

**Comment by tomitrescak** [comment]
@stubailo what's the substitute?

**Comment by tomitrescak** [comment]
@stubailo what's the substitute?

**Comment by lorensr** [comment]
@tomitrescak https://github.com/apollostack/meteor-integration/issues/18#issuecomment-230019721

**Comment by tomitrescak** [comment]
Nevermind, ... But I liked the central approach ... Call me old fashioned :)

**Comment by tomitrescak** [comment]
Nevermind, ... But I liked the central approach ... Call me old fashioned :)

**Comment by tomitrescak** [comment]
Nevermind, ... But I liked the central approach ... Call me old fashioned :)

**Comment by tomitrescak** [comment]
Nevermind, ... But I liked the central approach ... Call me old fashioned :)

**Comment by tomitrescak** [comment]
Nevermind, ... But I liked the central approach ... Call me old fashioned :)

**Comment by lorensr** [comment]
Weird, think your browser got into a submit loop 😆

image

**Comment by tomitrescak** [comment]
Oh, sh.t I'm in Eastern Europe in a mountain  cottage on Edge connection, things always get interesting  this way I guess. Sorry! Going offline now.

**Pull request closed without merging**